### PR TITLE
[MAINT] Update to new Python and dev tools

### PR DESCRIPTION
**Pull request by adam2392**
Thanks for contributing. If this is your first time,
make sure to read [contributing.md](https://github.com/mne-tools/mne-hfo/blob/master/CONTRIBUTING.md)

PR Description
--------------

- Ran black
- Renamed `docs/` -> `doc/`
- Setup `pyproject.toml` 
- Updated min Python version to 3.8 - 3.11
- Updated GH actions and circleCI workflows

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-hfo/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Comment by adam2392** [comment]
@drammock @larsoner any chance either of you know how to get the circleCI to be able to git pull from this repo? This works fine on `mne-icalabel` and `mne-connectivity`.

**Comment by larsoner** [comment]
Looks like https://discuss.circleci.com/t/git-github-com-permission-denied-publickey-on-repo-that-previously-worked/37763

**Comment by larsoner** [comment]
I followed the advice in that thread, namely clicking "Stop Building" in the project in the CircleCI project settings page, then re-added it under the "projects" tab of the mne-tools org on CircleCI and now the `checkout` step succeeds. I'll let you continue :)

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# YAML 1.2
+---
+# Metadata for citation of this software according to the CFF format (https://citation-file-format.github.io/)
+cff-version: 1.2.0
+title: 'MNE-HFO: Estimate HFOs with MNE in Python'
+authors:
+    - given-names: Adam
+      family-names: Li
+      affiliation: 'Department of Computer Science, Columbia University, New York, NY, USA'
+      orcid: 'https://orcid.org/0000-0001-8421-365X'
```
Comment (adam2392) [inline CITATION.cff:10]: @pmyers16 can you add an entry here for you?

**Pull request merged**